fotoapparat/src/main/java/io/fotoapparat/view/TextureRendererView.java

```diff
@@ -117,11 +114,41 @@ private void awaitSurfaceTexture() {
 
     @Override
     protected void onLayout(boolean changed, int left, int top, int right, int bottom) {
-        if (previewSize == null) {
+        if (rendererParameters == null) {
             super.onLayout(changed, left, top, right, bottom);
             return;
         }
 
+        Size previewSize = toPreviewSize(rendererParameters);
+
+        if (rendererParameters.scaleType == ScaleType.CENTER_INSIDE) {
```

Comment:
I was thinking, how about instead of passing scale type as the parameter of Fotoapparat instance we'll instead make it a parameter of the view which can be defined in XML? In the end, this parameter only changes the behavior of this particular view and not anything else.